### DISCUSSION TITLE
Phone & password authentication

### DISCUSSION THREAD

**Discussion by paradoxe113** [supabase/supabase#29568] [Questions]
Hi everyone, anyone who Can help. I've used Twilio not Twilio Verify(because I was getting the error message 'missing required parameter To in the body') so I gave up And finally used Twilio as Phone provider. The thing is, it uses Message Service it doesnt send OTP so I disabled phone Confirmation, it just send a Message To a customer and in the body part of the message I Can manipulate it and generate a code that I store Somewhere for later verification, so I handle the verification by myself. Everything is fine but when I want To register the user with the supabase.auth.signUp method and passing the phone and password I got an error, Internal error can't signUp New user in Database. Can you Please help. I Can share some snippets if needded I really need help. I'm working on an angular project. ![IMG_20240928_104711.jpg](https://github.com/user-attachments/assets/4d5333d7-d403-4f98-9621-332243e40d31)

**Comment by bytaesu** [comment]
Could you confirm whether the phone number is being passed in the international format? For instance, it should start with a “+” followed by the country code and then the phone number (e.g., +821055556666)

**Reply by paradoxe113** [reply]
I already do receive the message, the problem is To register the user After my verification, I am not in front of my pc but there are some screenshots ![IMG_20240928_135855.jpg](https://github.com/user-attachments/assets/bd5af951-583d-412e-9ee3-e53662c6446b)

![Screenshot_2024-09-28-14-01-05-525_com.foxdebug.acodefree.jpg](https://github.com/user-attachments/assets/bae63ebd-bc52-4578-b39f-e2ae3442ce35)

**Reply by bytaesu** [reply]
I would like to see the verification logic where supabase.auth.verifyOtp is used.

If it’s handled appropriately at the endpoint (In the case of PKCE flow), there shouldn’t be any issues.

Please share the information when you’re available on your PC. I am using Next.js, but I will try to help.

**Reply by paradoxe113** [reply]
Thanks man I really appreciate... I'm actually not using supabase.auth.VerifyOtp, I generated the content of the message and the 6 digit code by myself. I'll give you more details when I get To work, sorry it's kind of my rest day today and tomorrow

**Reply by bytaesu** [reply]
😎👌
